FEATURE - 무한스크롤 구현 수정

eunbbori opened the pull request:
### Background
- server와의 통신 후에 무한스크롤이 제대로 구현되지 않음

### Desc
- useCallback을 활용해 pageCount가 늘어나는 시점에 loadPostList action 호출

### Ticket link
-

eunbbori committed:
FEATURE - 무한스크롤 구현 수정

### Background
- server와의 통신 후에 무한스크롤이 제대로 구현되지 않음

### Desc
- useCallback을 활용해 pageCount가 늘어나는 시점에 loadPostList action 호출

KSH-code commented on client/src/components/Post/PostList.tsx:
```diff
@@ -1,54 +1,43 @@
+  useEffect(() => {
+    const handleScroll = () => {
+      const {scrollHeight, scrollTop, clientHeight } = document.documentElement;
+      if (scrollTop + clientHeight +100 >= scrollHeight) {
```
```suggestion
      if (scrollTop + clientHeight + 100 >= scrollHeight) {
```
lint 적용해야되겠군요

KSH-code commented on client/src/components/Post/PostList.tsx:
```diff
@@ -1,54 +1,43 @@
-    if (fetching) return
+  useEffect(() => {
+    if (fetching) fetchArticle()
+    else if (pageCount >= MAXIMUM_PAGES) setFetching(false)
```
MAXIMUM_PAGES에 도달하면 pageCount올리면서 데이터 더 불러올 필요 없으므로 removeEventListener 하는게 더 깔끔한 거 같기도 하네요

KSH-code commented on client/src/components/Post/PostList.tsx:
```diff
@@ -1,54 +1,43 @@
-    setFetching(true)
+  const fetchArticle = useCallback(()=> {
+    dispatch(allAction.loadPostList());
     setPosts(fetchedPosts)
```
컴포넌트 내에서 쓰는 posts와 fetchedPosts 가 혼용되어서 잘못 사용되고 있는거 같은데, 한 쪽으로 통일해서 사용하면 깔끔할 거 같아요

eunbbori replied:
앗 수정해야겠네요!ㅎㅎ

eunbbori committed:
Merge branch 'main' into feature/ec-04-get-post-view

eunbbori committed:
refactor

### Desc
- fetching, setFetching 변수명 조금 더 명확한 의미를 담게 fetchingPost,setFetchingPost 로 수정
- 코드리뷰 반영해 불필요한 중복 상태 제거
- pageCount >= MAXIMUM_PAGES일 경우 데이터 더 불러올 필요 없으므로 removeEventListener

eunbbori commented:
code 수정 후 다시 pull request 하겠습니다
